### PR TITLE
KOGITO-4336 - New persistence commons module for PostgreSQL based per…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…sistence using JsonB

### CODE REVIEW

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by radtriste — APPROVED** [review]
lgtm

**Review by nmirasch — APPROVED** [review]

File: persistence-commons/persistence-commons-postgresql/src/main/java/org/kie/kogito/persistence/postgresql/PostgresStorage.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    @Override
+    public void addObjectRemovedListener(Consumer<String> consumer) {
+        throw new UnsupportedOperationException(LISTENER_NOT_AVAILABLE_IN_POSTGRES_SQL);
+    }
+
+    @Override
+    public Query<V> query() {
+        throw new UnsupportedOperationException("Custom query not available in PostgresSQL");
```
Comment (manstis) [inline persistence-commons/persistence-commons-postgresql/src/main/java/org/kie/kogito/persistence/postgresql/PostgresStorage.java:75]: @cristianonicolai Hello, are there plans to add support for queries? Trusty uses queries and I am now writing a postgresql persistence implementation for it.. lack of query support is going to bite me.

Comment (cristianonicolai) [reply persistence-commons/persistence-commons-postgresql/src/main/java/org/kie/kogito/persistence/postgresql/PostgresStorage.java:75]: it needs to be done, but not sure if we have bandwidth in the near term.

**Pull request merged**